### DISCUSSION TITLE
Build speed of "rust-web-template" in "wasm pack" environment is too slow.

### DISCUSSION THREAD

**Discussion by dolto** [bevyengine/bevy#10287] [Q&A] [answered]
Hello I'm a novice developer trying this and that in a web environment with "bevy".

In the past, I didn't really think about the time it would take to build (though that was an average of 1 to 3 minutes)

However, it started to take more than six minutes to build once because the scale of trying is getting bigger and using various external plug-ins.

So to reduce compilation time, I looked here and there, and I heard there are several ways on the official website, so I checked and found the following.

![image](https://github.com/bevyengine/bevy/assets/43398206/82be134b-f053-4b7c-8381-ec9687aa647f)
```
[dependencies]
bevy = { version = "0.11.0", features = ["dynamic_linking"] }
``` 

And i also installed "LLD linker" as explained on the homepage and added it to the Toml file as follows.
(Unfortunately, there was a problem when I modified the .cargo/config.toml file, so I didn't modify it. T_T )

```
[toolchain]
channel = "nightly"
``` 

Anyway, I tried that, and the following error occurred.

```
error: cannot produce dylib for `bevy_dylib v0.11.3` as the target `wasm32-unknown-unknown` does not support these crate types
``` 

As a result of the investigation, I tried it because it could work if I tried as follows.

After looking into it, we found that the 0.8.0 version also had the following problems, and we tried the method presented there.

```
[dependencies]
bevy = { version = "0.11.3", default-features = false, features = ["bevy_winit", "dynamic_linking"] }
#...
[target.'cfg(target_arch = "wasm32")'.dependencies]
console_log = { version = "1.0.0" }
console_error_panic_hook = { version ="0.1.7" }
wee_alloc = { version = "0.4.5" }
``` 

But that method didn't work either.

What can I try now?

**Comment by nicopap** [comment, marked as answer]
Dynamic linking doesn't work with web target.

Compile time for WASM is a pretty big problem in rust. I've been annoyed by it myself, and I'm not aware of a good solution. The way I do it is by testing the desktop version and then only compile to WASM when publishing (for example, publishing on itch.io). There isn't much you need to do to make your bevy game run on WASM, so when it comes time to compile to WASM, it's not that bad if you've been only testing on native, the changes you have to make are usually easy to find and fix.

**Reply by dolto** [reply]
That's sad news.
Strangely, "cargo run" didn't work in the wasm-pack, so I need to find out more about this.

Thank you.
